### PR TITLE
fix: Form value fix for spend date

### PR DESCRIPTION
**Pull request by suyashpatil78**
### Description
copilot:summary

copilot:poem

### Walkthrough
copilot:walkthrough

app.clickup.com

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/split-expense/split-expense.page.ts [hunk 1]
```diff
@@ -975,13 +1032,6 @@ export class SplitExpensePage {
     this.add(amount1, this.currency, percentage1, null);
     this.add(amount2, this.currency, percentage2, null);
     this.getTotalSplitAmount();
-
-    const today = new Date();
```
Comment (suyashpatil78) [inline src/app/fyle/split-expense/split-expense.page.ts:979]: this should not be `setValuesForCCC` as it is irrelevant to the function

File: src/app/fyle/split-expense/split-expense.page.ts [hunk 2]
```diff
@@ -888,6 +940,11 @@ export class SplitExpensePage {
     this.selectedProject = JSON.parse(this.activatedRoute.snapshot.params.selectedProject as string) as ExtendedProject;
     this.expenseFields = JSON.parse(this.activatedRoute.snapshot.params.expenseFields as string) as ExpenseField[];
 
+    // Set max and min date for form
+    const today = new Date();
```
Comment (suyashpatil78) [inline src/app/fyle/split-expense/split-expense.page.ts:944]: this is the same logic as in add-edit-expense

**Review by Chethan-Fyle — APPROVED** [review]

File: src/app/fyle/split-expense/split-expense.page.ts [hunk 3]
```diff
@@ -240,15 +243,58 @@ export class SplitExpensePage {
...
+    txnDate.setMinutes(0);
+    txnDate.setSeconds(0);
+    txnDate.setMilliseconds(0);
+    txnDate = this.timezoneService.convertToUtc(txnDate, offset);
+
+    return txnDate;
+  }
+
+  correctDates(transaction: Transaction, offset: string): void {
+    // setting from_date and to_date time to T10:00:00:000 in local time zone
```
Comment (Dimple16) [inline src/app/fyle/split-expense/split-expense.page.ts:267]: I guess we have a similar logic added in one of the methods of the Data service - can we reuse it?

**Review by Dimple16 — APPROVED** [review]
minor comments - please fix them before merging

File: src/app/fyle/split-expense/split-expense.page.ts (hunk 3 above)
Comment (suyashpatil78) [reply src/app/fyle/split-expense/split-expense.page.ts:267]: I guess you are referring to this - https://github.com/fylein/fyle-mobile-app/blob/master/src/app/core/services/transaction.service.ts#L292
But here, there are other operations too with this logic

**Pull request merged**